### PR TITLE
Feature: Support placeholder images (defaultSource prop)

### PR DESCRIPTION
**Pull request by IjzerenHein**
Hi, this PR adds support for placeholder images, which are shown until the image is loaded, or when the image cannot be loaded at all (e.g. when a bad storage path is provided). This adds a prop called `defaultSource` which follows the same API as the react-native Image `defaultSource` prop does. Usage is as follows:

```
<ImageView path='...' defaultSource={require('../assets/placeholder.png')} ... />
```

It then looks like this when an image cannot be loaded:

image

The following changes/fixes have been made to make this possible:
- Native support for defaultSource prop on Android
- Native support for defaultSource prop on iOS
- Moved JavaScript code to `src/` and added a separate file for Android, in order to support the new defaultSource prop
- The code for the defaultSource prop is nearly identical to that of the standard RN Image implementation
- Fixed a linker issue on iOS due to a newer version of the GoogleUtilities pod
- Made it possible to build signed APK releases (using the debug keystore) of the example app, so it was possible to verify that defaultSource works on Android (see caveat below)
- Extended the Example app with a placeholder image and it now uses the `defaultSource` prop
- Updated the README with the new `defaultSource` prop

Caveats:
- On Android, the defaultSource prop does not work on debug builds. This is a limitation of the Android image handling implementation. This behavior is identical to that of the standard react-native Image component, which also has this limitation. (see https://facebook.github.io/react-native/docs/image#defaultsource)

cheers,
Hein

### CODE REVIEW

**Comment by erezrokah** [comment]
Thanks @IjzerenHein, great work. I'll go over it soon

File: src/index.android.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// On Android, both Image and Photo view are supported,
+// and defaultSource is converted to an uri that the
```
Comment (erezrokah) [inline src/index.android.js]: I think it should be "a uri", but not a big thing (I can fix it after the merge)

**Comment by erezrokah** [comment]
I'm just referencing some code for clarity:

Android: https://github.com/facebook/react-native/blob/7795a672d3a24a5b50df6ad6d30555d856b557cc/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java#L376

iOS: https://github.com/firebase/FirebaseUI-iOS/blob/ff1b8afd2835204e85752d49f25a29c632a42a77/Storage/FirebaseStorageUI/UIImageView%2BFirebaseStorage.m#L89

File: src/index.android.js (hunk 1 above)
Comment (IjzerenHein) [reply src/index.android.js]: I think you're right :)

**Comment by IjzerenHein** [comment]
> I'm just referencing some code for clarity:
> 
> Android: https://github.com/facebook/react-native/blob/7795a672d3a24a5b50df6ad6d30555d856b557cc/ReactAndroid/src/main/java/com/facebook/react/views/image/ReactImageView.java#L376
> 
> iOS: https://github.com/firebase/FirebaseUI-iOS/blob/ff1b8afd2835204e85752d49f25a29c632a42a77/Storage/FirebaseStorageUI/UIImageView%2BFirebaseStorage.m#L89

Good idea!

**Pull request merged**